js/core.js

```diff
@@ -1162,6 +1162,36 @@ async function parsePhraseBook(s, loadSketch) {
             phraseBook[phrase.key].parameters = phrase.parameters;
         }
     }
+
+    let references = [];
+    function traverse(x){
+        if(x.type === "Concat"){
+            for(key in x){
+                if(x[key].type){
+                    traverse(x[key]);
+                }
+            }
+        }
+        else if(x.type === "Ref"){
+            if(references.indexOf(x.node.key) === -1){
+                references[references.length] = x.node.key;
+            }
+        }
+    }
+    phrases.forEach(function(item){
+        if(item.key === "root"){
+            item.values.forEach(function(i){
+                let p = parsePhrase(i);
+                traverse(p);
+            });
+        }
+    });
+    references.forEach(function(item){
+        if(!phraseBook[item]){
```

Comment:
Also, a key can also be a reference to something stored in localMemory, like this:

```root:
- {{ dosomething(123) }}

dosomething(num):
- I have to do {{ num }} things.```

num is also a key but doesn't refer to a phrase, but in your check it will throw an error.